app/main.py

```diff
@@ -1,2 +1,87 @@
+from dataclasses import dataclass
+from typing import Hashable, Iterator
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: any
+    hash_value: int
+
+
 class Dictionary:
-    pass
+    INITIAL_CAPACITY = 8
+    THRESHOLD = 2 / 3.0
+    CAPACITY_MULTIPLIER = 2
+
+    def __init__(self, initial_capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.buckets = [[] for _ in range(self.capacity)]
+
+    def _get_index(self, key: Hashable) -> int:
+        return hash(key) % self.capacity
+
+    def _resize(self) -> None:
+        new_capacity = self.capacity * self.CAPACITY_MULTIPLIER
+        new_buckets = [[] for _ in range(new_capacity)]
+
+        for bucket in self.buckets:
+            for node in bucket:
+                new_index = node.hash_value % new_capacity
+                new_buckets[new_index].append(node)
+
+        self.capacity = new_capacity
+        self.buckets = new_buckets
+
+    def __setitem__(self, key: Hashable, value: any) -> None:
+        hash_key = hash(key)
+        index = self._get_index(key)
+        bucket = self.buckets[index]
+
+        for node in bucket:
+            if node.key == key:
+                node.value = value
+                return
+
+        new_node = Node(key=key, value=value, hash_value=hash_key)
+        bucket.append(new_node)
+        self.size += 1
+
+        load_factor = self.size / self.capacity
+        if load_factor > self.THRESHOLD:
```

Comment:
The condition `if load_factor > self.THRESHOLD:` is correct for triggering a resize. Ensure that the `THRESHOLD` value is set appropriately to avoid frequent resizing or excessive memory usage.